dev_docs/contributing/standards.mdx

```diff
@@ -15,6 +15,45 @@ We expect all developers to read and abide by our overarching <DocLink id="kibDe
 
 Please read and abide by our <DocLink id="kibStyleGuide" text="Style guide" />. The majority of these items are linted against but some are not.
 
+## Adding dependencies
+
+Looking for a dependency that isn't already available in Kibana? There are a few things to keep in mind before adding a new dependency.
+
+First, be sure you have read and are familiar with our <DocLink id="kibDevPrinciples" />. In particular, **Be wary of dependencies**
+and **Prefer one way to do things** provide an overview of how we approach this question.
+
+In general, we have a bias toward **not** adding new dependencies unless there is a compelling reason to do so, as we want to
+minimize Kibana's overall complexity.
+
+Should you find yourself evaluating a new dependency, here are some specific things to ask yourself:
+
+1. **Is there already another dependency that offers similar functionality?** If so, adding another dependency may not be necessary.
+Prefer one way to do things and use what's already there, unless there is an important reason not to do so.
+2. **Does this dependency appear to be well-maintained?** A dependency that hasn't been updated in years is usually more of a
+liability than an asset.
+3. **Does this dependency have a license that's compatible with Kibana's?** Most common open source licenses such as BSD, MIT,
+and Apache 2.0/1.1 are okay to use with Kibana. Others may not be, or may require special attribution.
+4. **Will this dependency need to be prebuilt?** Due to our build process, native module dependencies should include at least a prebuild
+step so at install time it simply downloads instead of building from source. This allows us to optimize bootstrap times.
+5. **Am I committed to maintaining this dependency?** Once you add a dependency to the `package.json`, someone else isn't going to
+keep it updated for you. That means you will be responsible for updating it regularly, keeping an eye out for security vulnerabilities,
+and dealing with any breaking changes that may arise during an upgrade. We recommend relying on the renovate bot to help keep the
+dependency updated; be sure to mark your ownership of the package in the
+[`renovate.json`](https://github.com/elastic/kibana/blob/main/renovate.json`) file.
```

Comment:
nit: Consider adding some language and/or bullet point around size of the package?

e.g. If the dependency is large (some arbitrary MB size), please think twice whether to add it.

The reason for this is that _size_ in and of itself has been part of the motivation for us to take on dev-work internally (e.g. Adding flamegraph to elastic/charts, since 3rd party dependencies were quite bloated).

Reply:
Good idea, I'll add something to that effect